src/main/java/org/threadly/concurrent/SingleThreadScheduler.java

```diff
@@ -246,29 +247,40 @@ protected void finalize() {
    * @author jent - Mike Jensen
    * @since 2.0.0
    */
-  protected static class SchedulerManager extends AbstractService 
-                                          implements Runnable {
+  protected static class SchedulerManager implements Runnable {
     protected final NoThreadScheduler scheduler;
+    protected final AtomicInteger state = new AtomicInteger(-1); // -1 = new, 0 = started, 1 = stopping, 2 = stopped
```

Comment:
I think its fine, though I dont really see the point of starting at -1, but its fine.